### ISSUE TITLE
IOS build error after adding Firebase Dynamic Links plugin

### ISSUE THREAD

**Issue by matheuskt** [andrehtissot/cordova-plugin-fcm-with-dependecy-updated#44]
I've been using this plugin with no problem for a while. Last week I added the Firebase Dynamic Links plugin to my Ionic app and wasn't able to build for Android. It was throwing this error while trying to build:

```
Dependency failing: com.google.firebase:firebase-messaging:19.0.0 -> com.google.firebase:firebase-iid@[19.0.0], but fire
  base-iid version was 20.0.2.
```

After some research I found a solution. Adding androidx and androidx-adapter plugins plus changing the version to **20.1+** on the following line

```
cordova.system.library.9=com.google.firebase:firebase-messaging:18.0
```
on the **project.properties** file, this for Android.

But now I'm trying to build for IOS and it's throwing me errors for this plugin again:

```
The following build commands failed:
	CompileC /Users/user911231/Library/Developer/Xcode/DerivedData/Project-fwezigjjyvjbvcgqjcyweuybexmt/Build/Intermediates.noindex/Project.build/Release-iphonesimulator/Project.build/Objects-normal/x86_64/FCMPlugin.o /Users/user911231/Desktop/Projects/Project/project-mobile/platforms/ios/Project/Plugins/cordova-plugin-fcm-with-dependecy-updated/FCMPlugin.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler

xcodebuild: Command failed with exit code 65
```

 I believe it's conflicting with the recently added plugin since that if I remove dynamic links on both platforms the build is successful. Is there anything I can do similar to what I did above for Android? What could be the reason?

**Comment by andrehtissot** [comment]
There should be, but it is strange that the error was on build time.
Would you:
1. Try to make a clean build, to be sure it's not a cache issue;
2. Create a hello world repository were I can debug this issue, with the exact version that you are using? Or maybe share you current one if open source.

**Comment by matheuskt** [comment]
Got it working this morning. 

Turns out the error message was thrown a bit above on the log. It was the Firebase.h file that was missing.

Updated cordova to 9.0 (previously at 8.0) and the project plugins, but the build kept on failing with the error this time being conflicts between the plugins' dependencies.

After that I had to run ```pod update && pod install``` inside the ```platforms/ios``` folder.

Didn't catch the real error messaging before opening this issue, if I did I wouldn't have opened it, sorry for this. I'm closing this now.

Thanks for creating/maintaining this plugin and for the quick response.
